OpenRA.Mods.Common/Effects/Bullet.cs

```diff
@@ -132,11 +130,32 @@ public void Tick(World world)
 				smokeTicks = info.TrailInterval;
 			}
 
+			var nextpos = WPos.LerpQuadratic(args.Source, target, angle, ticks, length);
+
+			if (!info.High)
+			{
+				var cell = world.Map.CellContaining(pos);
+				var nextcell = world.Map.CellContaining(nextpos);
+
+				var regularLength = (cell - nextcell).LengthSquared;
+
+				var collision = world.ActorMap.GetActorsOnRay(cell, nextcell).Where(x => x.HasTrait<IBlocksBullets>())
+					.FirstOrDefault(actor => !actor.IsDead && regularLength >= (cell - actor.Location).LengthSquared);
+
+				if (collision != null)
+				{
+					pos = collision.CenterPosition;
```

Comment:
I mean `map.CenterOfCell(collisionCell)`.  To get that cell you'll need to run the raytracer directly from here (instead of via ActorMap), and then enumerate over the cells and actors so you have access to both.